Query  using the Or  operator do not provide .iter() method, only iter_mut().

**Bevy version**
Following version, containing changes on query:
9cc6368b28a2df2dff652ae1a087f74fd0362a6a

**Operating system & version**
Ubuntu 20.04

**What you did**
The following query, that contains an Or, does not have the .iter() method, only iter_mut()
```rust 
mut query_changed_camera: Query<(
        &Camera,
        Or<(Changed<OrthographicProjection>, Changed<Transform>)>,
    )>,
```
```rust
for (camera, (camera_ortho, camera_transform)) in query_changed_camera.iter() {
```

**What you expected to happen**
Should have compiled

**What actually happened**
Compilation failed:
```rust 
error[E0277]: the trait bound `bevy_hecs::query::FetchOr<(bevy_hecs::query::FetchChanged<bevy::render::camera::OrthographicProjection>, bevy_hecs::query::FetchChanged<bevy::prelude::Transform>)>: bevy::ecs::ReadOnlyFetch` is not satisfied
  --> src/cursor_2dworld_pos_plugin.rs:62:76
   |
62 |     for (camera, (camera_ortho, camera_transform)) in query_changed_camera.iter() {
   |                                                                            ^^^^ the trait `bevy::ecs::ReadOnlyFetch` is not implemented for `bevy_hecs::query::FetchOr<(bevy_hecs::query::FetchChanged<bevy::render::camera::OrthographicProjection>, bevy_hecs::query::FetchChanged<bevy::prelude::Transform>)>`
   |
   = note: required because of the requirements on the impl of `bevy::ecs::ReadOnlyFetch` for `(bevy_hecs::query::FetchRead<bevy::render::camera::Camera>, bevy_hecs::query::FetchOr<(bevy_hecs::query::FetchChanged<bevy::render::camera::OrthographicProjection>, bevy_hecs::query::FetchChanged<bevy::prelude::Transform>)>)`
```

**Additional information**

Using the .iter_mut() method works, so, nothing critical here.

As discussed in the discord, the culprit is the 'Or' , not the Changed.

it can be the same issue as #753 , but we were not sure on discord, so filing another issue.

--- a/crates/bevy_ecs/hecs/src/query.rs
+++ b/crates/bevy_ecs/hecs/src/query.rs
@@ -263,6 +263,9 @@ macro_rules! impl_or_query {
                 true $( && $T.should_skip(n) )+
             }
         }
+
+        unsafe impl<$( $T: ReadOnlyFetch ),+> ReadOnlyFetch for Or<($( $T ),+)> {}
+        unsafe impl<$( $T: ReadOnlyFetch ),+> ReadOnlyFetch for FetchOr<($( $T ),+)> {}
     };
 }
 
@@ -320,6 +323,7 @@ impl<'a, T: Component> Query for Mutated<'a, T> {
 
 #[doc(hidden)]
 pub struct FetchMutated<T>(NonNull<T>, NonNull<bool>);
+unsafe impl<T> ReadOnlyFetch for FetchMutated<T> {}
 
 impl<'a, T: Component> Fetch<'a> for FetchMutated<T> {
     type Item = Mutated<'a, T>;

--- a/crates/bevy_ecs/src/system/into_system.rs
+++ b/crates/bevy_ecs/src/system/into_system.rs
@@ -490,6 +490,43 @@ mod tests {
     }
 
     #[test]
+    fn or_query_set_system() {
+        // Regression test for issue #762
+        use crate::{Added, Changed, Mutated, Or};
+        fn query_system(
+            mut ran: ResMut<bool>,
+            set: QuerySet<(
+                Query<Or<(Changed<A>, Changed<B>)>>,
+                Query<Or<(Added<A>, Added<B>)>>,
+                Query<Or<(Mutated<A>, Mutated<B>)>>,
+            )>,
+        ) {
+            let changed = set.q0().iter().count();
+            let added = set.q1().iter().count();
+            let mutated = set.q2().iter().count();
+
+            assert_eq!(changed, 1);
+            assert_eq!(added, 1);
+            assert_eq!(mutated, 0);
+
+            *ran = true;
+        }
+
+        let mut world = World::default();
+        let mut resources = Resources::default();
+        resources.insert(false);
+        world.spawn((A, B));
+
+        let mut schedule = Schedule::default();
+        schedule.add_stage("update");
+        schedule.add_system_to_stage("update", query_system.system());
+
+        schedule.run(&mut world, &mut resources);
+
+        assert!(*resources.get::<bool>().unwrap(), "system ran");
+    }
+
+    #[test]
     fn changed_resource_system() {
         fn incr_e_on_flip(_run_on_flip: ChangedRes<bool>, mut i: Mut<i32>) {
             *i += 1;

